src/cinder/gl/Texture.cpp

```diff
@@ -1673,6 +1672,20 @@ void Texture3d::update( const void *data, GLenum dataFormat, GLenum dataType, in
 {
 	ScopedTextureBind tbs( mTarget, mTextureId );
 	glTexSubImage3D( mTarget, mipLevel, xOffset, yOffset, zOffset, width, height, depth, dataFormat, dataType, data );
+
+	if( mMipmapping && mipLevel == 0 ) {
+		glGenerateMipmap( mTarget );
+	}
+}
+
+void Texture3d::initMaxMipmapLevel()
+{
+	if( mMaxMipmapLevel == -1 ) {
+		mMaxMipmapLevel = requiredMipLevels( mWidth, mHeight, mDepth ) - 1;
+#if ! defined( CINDER_GL_ES_2 )
```

Comment:
Probably want to cover any ES implementations here. I ran into trouble (all-black textures) with ES3 on tvos.